### PR TITLE
Enable `--serve-artifacts` by default

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Enable `--serve-artifacts` by default when running a tracking server.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by WeichenXu123 — APPROVED** [review]

**Review by dbczumar — COMMENTED** [review]
Two small pieces of feedback:

1. When I run with `--backend-store-uri sqlite:///mydb.sqlite --no-serve-artifacts` and forget to specify `--default-artifact-root, I see:

```
mlflow server --backend-store-uri sqlite:///mydb.sqlite --no-serve-artifacts
Option 'default-artifact-root' is required when backend store is not local file based.
Usage: mlflow server [OPTIONS]
Try 'mlflow server --help' for help.

Error: Option 'default-artifact-root' is required when backend store is not local file based.
```

Can we update this to say that 'default-artifact-root' is required when backend store is not local file based and artifact serving is disabled?

2. Do we need to update docs, such as https://mlflow.org/docs/latest/tracking.html#using-the-tracking-server-for-proxied-artifact-access, to indicate that proxied artifact access is enabled by default and also talk about how the `--no-serve-artifacts` option can be used to opt out?

File: mlflow/utils/cli_args.py [hunk 1]
```diff
@@ -202,13 +202,13 @@ def _create_env_manager_option(help_string):
 )
 
 SERVE_ARTIFACTS = click.option(
-    "--serve-artifacts",
+    "--serve-artifacts/--no-serve-artifacts",
```
Comment (dbczumar) [inline mlflow/utils/cli_args.py:205]: Can we explain the behavior when `--no-serve-artifacts` is specified?

Comment (harupy) [reply mlflow/utils/cli_args.py:205]: Sure!

**Comment by harupy** [comment]
> Can we update this to say that 'default-artifact-root' is required when backend store is not local file based and artifact serving is disabled?

Got it, thanks for the catch!

> 2. Do we need to update docs, such as https://mlflow.org/docs/latest/tracking.html#using-the-tracking-server-for-proxied-artifact-access, to indicate that proxied artifact access is enabled by default and also talk about how the `--no-serve-artifacts` option can be used to opt out?

We do :) I'll update them.

**Review by dbczumar — APPROVED** [review]
LGTM once docs are updated; thanks @harupy !

File: docs/source/tracking.rst [hunk 2]
```diff
@@ -156,6 +160,10 @@ a Postgres database for backend entity storage, and an S3 bucket for artifact st
 
 .. figure:: _static/images/scenario_4.png
```
Comment (harupy) [inline docs/source/tracking.rst:161]: I need to update the images.

File: docs/source/tracking.rst [hunk 3]
```diff
@@ -132,6 +132,11 @@ server running, the MLflow client interacts with the tracking server via REST re
 
 .. figure:: _static/images/scenario_3.png
 
+.. code-block:: bash
+    :caption: Command to run the tracking server in this configuration
+
+    mlflow server --backend-store-uri file::///path/to/mlruns --no-serve-artifacts
```
Comment (harupy) [inline docs/source/tracking.rst:138]: Separated the command from the image to make it easier to edit and copy.

Comment (dbczumar) [reply docs/source/tracking.rst:138]: Fantastic!

```suggestion
.. code-block:: bash
    :caption: Command to run the tracking server in this configuration

    mlflow server --backend-store-uri file:///path/to/mlruns --no-serve-artifacts
```

File: docs/source/tracking.rst [hunk 4]
```diff
@@ -156,6 +161,11 @@ a Postgres database for backend entity storage, and an S3 bucket for artifact st
 
 .. figure:: _static/images/scenario_4.png
 
+.. code-block:: bash
+    :caption: Command to run the tracking server in this configuration
+
+    mlflow server --backend-store-uri postgresql:://URI --default-artifact-root s3://bucket_name --host remote_host --no-serve-artifacts
```
Comment (dbczumar) [inline docs/source/tracking.rst:167]: I think we should remove the double colons in these examples, right?

```suggestion
    mlflow server --backend-store-uri postgresql://URI --default-artifact-root s3://bucket_name --host remote_host --no-serve-artifacts
```

Comment (dbczumar) [reply docs/source/tracking.rst:167]: Can we also replace `URI` with a realistic example?

File: docs/source/tracking.rst (hunk 3 above)
Comment (harupy) [reply docs/source/tracking.rst:138]: Thanks for the catch!

**Review by dbczumar — APPROVED** [review]
LGTM once remaining comments are addressed! Thanks @harupy !

File: docs/source/tracking.rst (hunk 4 above)
Comment (harupy) [reply docs/source/tracking.rst]: Done!

**Pull request merged**